### PR TITLE
Update jemalloc to version 5.1.0

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
[jemalloc 5.1.0](https://github.com/jemalloc/jemalloc/releases/tag/5.1.0) has an impressive list of fixes (especially for non-x86 archs) and possible performance improvements.

**Comment by smuzaffar** [comment]
let it run for couple of DEVEL IBs first ( https://github.com/cms-sw/cmsdist/pull/4087 )

**Comment by fwyzard** [comment]
Here is a comparison of the time per event for the latest HLT menu with jemalloc 4.5.0, 5.0.1 and 5.1.0, running with CMSSW 10.1.4 over events with 50 pileup:

| version | time per event | job throughput (4 threads) |
| --- | --- | --- |
| jemalloc 4.5.0 |  474 ± 2 ms/ev  |  8.36 ± 0.06 ev/s |
| jemalloc 5.0.1 |  463 ± 3 ms/ev  |  8.56 ± 0.07 ev/s |
| jemalloc 5.1.0 |  463 ± 3 ms/ev  |  8.56 ± 0.07 ev/s |

Bottom line: we do want 5.0.1 (which is already in CMSSW 10.2.x), while there is no reason to push for 5.1.0 from the performance point of view.

**Comment by fwyzard** [comment]
**Update:** I have also tried different MALLOC_CONF options, but I observed only negligible changes:

|                                            |  total time/event |  throughput (4 threads) |
| ------------------------------------------ | ----------------- | ----------------------- |
| jemalloc_4.5.0,                            |        474 ± 2 ms |        8.36 ± 0.06 ev/s |
| jemalloc_5.0.1,                            |        463 ± 3 ms |        8.56 ± 0.07 ev/s |
| jemalloc_5.1.0,                            |        464 ± 3 ms |        8.56 ± 0.07 ev/s |
|                                            |                   |                         |
| jemalloc_5.1.0, cmssw  default             |        463 ± 3 ms |        8.58 ± 0.06 ev/s |
| jemalloc_5.1.0, empty options              |        463 ± 4 ms |        8.56 ± 0.07 ev/s |
| jemalloc_5.1.0, `background_thread:true`   |        464 ± 3 ms |        8.55 ± 0.05 ev/s |
| jemalloc_5.1.0, `metadata_thp:always`      |        465 ± 4 ms |        8.54 ± 0.06 ev/s |
| jemalloc_5.1.0, `metadata_thp:auto`        |        464 ± 3 ms |        8.57 ± 0.07 ev/s |

**Comment by smuzaffar** [comment]
thanks @fwyzard for the comparison. looks good to me

**Comment by smuzaffar** [comment]
please test
lets run a fresh test

**Comment by smuzaffar** [comment]
+externals
